SpatialGDK/Build/Programs/Improbable.Unreal.Scripts/DeploymentLauncher/DeploymentLauncher.cs

```diff
@@ -101,136 +102,47 @@ private static int CreateDeployment(string[] args)
             {
-                // Upload snapshots.
-                var mainSnapshotId = UploadSnapshot(snapshotServiceClient, mainDeploymentSnapshotFilePath, projectName,
-                    mainDeploymentName);
+                var deploymentServiceClient = DeploymentServiceClient.Create();
 
-                if (mainSnapshotId.Length == 0)
-                {
-                    return 1;
-                }
+                var createMainDeploymentOp = CreateMainDeploymentAsync(deploymentServiceClient, launchSimPlayerDeployment, projectName, assemblyName, mainDeploymentName, mainDeploymentJsonPath, mainDeploymentSnapshotPath);
 
-                // Create main deployment.
-                var mainDeploymentConfig = new Deployment
-                {
-                    AssemblyId = assemblyName,
-                    LaunchConfig = new LaunchConfig
-                    {
-                        ConfigJson = File.ReadAllText(mainDeploymentJson)
-                    },
-                    Name = mainDeploymentName,
-                    ProjectName = projectName,
-                    StartingSnapshotId = mainSnapshotId
-                };
-
-                mainDeploymentConfig.Tag.Add(DEPLOYMENT_LAUNCHED_BY_LAUNCHER_TAG);
-
-                if (launchSimPlayerDeployment)
+                if (!launchSimPlayerDeployment)
                 {
-                    // This tag needs to be added to allow simulated players to connect using login
-                    // tokens generated with anonymous auth.
-                    mainDeploymentConfig.Tag.Add("dev_login");
+                    // Don't launch a simulated player deployment. Wait for main deployment to be created and then return.
+                    Console.WriteLine("Waiting for deployment to be ready...");
+                    createMainDeploymentOp.PollUntilCompleted();
+                    Console.WriteLine("Successfully created the main deployment");
+                    return 0;
                 }
 
-                Console.WriteLine(
-                    $"Creating the main deployment {mainDeploymentName} in project {projectName} with snapshot ID {mainSnapshotId}. Link: https://console.improbable.io/projects/{projectName}/deployments/{mainDeploymentName}/overview");
-
-                var mainDeploymentCreateOp = deploymentServiceClient.CreateDeployment(new CreateDeploymentRequest
-                {
-                    Deployment = mainDeploymentConfig
-                }).PollUntilCompleted();
+                var createSimDeploymentOp = CreateSimPlayerDeploymentAsync(deploymentServiceClient, projectName, assemblyName, mainDeploymentName, simDeploymentName, simDeploymentJson, simNumPlayers);
 
-                Console.WriteLine("Successfully created the main deployment.");
+                // Wait for both deployments to be created.
+                Console.WriteLine("Waiting for deployments to be ready...");
+                createMainDeploymentOp.PollUntilCompleted();
```

Comment:
Should we add a failure check for the main deployment here?

Reply:
Yeah good point, will add